### ISSUE TITLE
Nested child schema pre-save hooks not working since version 5.5.4

### ISSUE THREAD

**Issue by paksu** [Automattic/mongoose#7792]
We are using nested schemas and we have a child schema that has a pre-save hook. After upgrading to latest mongoose the child schema pre save hooks stopped working. Seems that this started happening since mongoose 5.5.4. With mongoose 5.5.3 the hooks are still being called.

Code that you can use to replicate the issue
```const childSchema = new mongoose.Schema({ test: String });
  childSchema.pre('save', function () {
    console.log('HELLO WORLD')
  });

  const topSchema = new mongoose.Schema({
    children: {
      child1: childSchema,
      child2: childSchema,
    }
  });

  const Model = mongoose.model('Model', topSchema);

  const m = await new Model({
    children: {
      child1: {},
      child2: {},
    }
  }).save();
```
Before when calling `save` for the model the child schema pre-save hooks would run but with version 5.5.4 or later they are not being called.

**Comment by lasselukkari** [comment]
I can confirm that the behaviour has changed between the versions. 

Running this script on 5.5.3 prints the expected output. With 5.5.4 the pre save hook does not run. The `children.child.name` property is present in the saved document with both versions.
```js
const mongoose = require('mongoose');
mongoose.connect('mongodb://localhost:27017/test', { useNewUrlParser: true });

const childSchema = new mongoose.Schema({ name: String });

childSchema.pre('save', function () {
  console.log('Running pre save hook')
});

const topSchema = new mongoose.Schema({
  children: {
    child: childSchema
  }
});

const Model = mongoose.model('Model', topSchema);

new Model({ children: { child: { name: "Test" } } }).save(console.log);
```

**Comment by Fonger** [comment]
This definitely a priority issue as it breaks a core feature

I can confirm this regression is caused by the fix of #7719 
https://github.com/Automattic/mongoose/commit/d8c2618dc79e6f75b04fe3adde148e0ab77b9722

**Comment by vkarpov15** [comment]
Thanks for reporting this problem and sorry for the trouble. Fix will be in 5.5.8, which will be released within the next few hours.
